SPEC.md

```diff
@@ -98,61 +98,114 @@ A build tool is something which reads one or more input artifacts and writes one
+blob⎵${artifact id of child}\n
 ```
 
 A child artifact which is itself a derived artifact would be represented by
 ```
-blob⎵${git object id of child}⎵bom⎵${git object id of child's GitBOM document}\n
+blob⎵${artifact id of child}⎵bom⎵${GitBOM identifier of child's GitBOM Document}\n
 ```
 
-### GitBOM identifier
+The child artifact records must be written to the GitBOM Document in lexical order.
 
-For a given artifact, the GitBOM ID of that artifact is simply the git object id of it's GitBOM Document.
+The artifact id and GitBOM Document id must both be of the Gitoid Identifier Type declared in the GitBOM Document header.
 
 ### GitBOM Identifier Embedding
 
-Each build tool should embed the GitBOM ID of its output artifact(s) into the output artifact in a manner:
+Each build tool should embed into the output artifact a new line delimited, lexically ordered, list of GitBOM identifiers for each mandatory Gitoid Identifier Type in a manner:
 
-1.  Appropriate to the type of artifact
-2.  Generally agreed upon for that artifact
+1. Appropriate to the type of artifact
+2. Generally agreed upon for that artifact
 
 ### GitBOM Document Construction by a Build Tool
 
-When a build tool reads an artifact, it should for each input file:
-1. Compute the git object id of the input file - ${artifact identifier}
-2. Examine the artifact for an embedded GitBOM identifier - ${gitbom idenfitier}
-3. If the artifact contains a GitBOM identifer it construct a record:
-   ```
-   blob⎵${artifact identifier}⎵bom⎵${GitBOM identifier}\n
-   ```
-4. Otherwise construct a record:
-   ```
-   blob⎵${artifact identifier}\n
-   ```
-
-Concatenate all records together and apply lexical ordering by line to create the GitBOM document of the
-output artifact.
+A build tool creating an output artifact must compute a GitBOM Document of each mandatory artifact id type.
+
+For each input artifact the build tool must:
+
+1. Compute the git object id of the input - ${artifact identifier}
+2. Examine the input for an embedded GitBOM Document Identifier - ${gitbom document identifier}
+
+The build tool must persist a GitBOM Document using the ${artifact identifier} and ${gitbom document identifier} for each input.
 
 ### GitBOM Document persistence by a Build Tool
 
-When persisting an output artifact to a file system, the build tool should create a subdirectory of the directory to which the output artifact is being written of the form:
-```.bom/objects/``` and write out the GitBOM document to ```.bom/objects/${gitoid:0:2}/${gitoid:2:}``` where ```${gitoid}```
-is the git object id of the GitBOM document.
+Each build tool should persist all GitBOM Documents it generates.
+
+Generically, GitBOM persistence can be thought of as having an abstract API:
+```
+WriteGitBOM(GitBOM id,GitBOM contents) returns error
+ReadGitBOM(GitBOM id) returns GitBOM Contents
+```
+
+WriteGitBOM should check that GitBOM id matches GitBOM contents and return an error if it does not.
```

Comment:
```suggestion
```
I suggest removing mention of error checking within the function, as that is ancillary to the format being specified here.

Reply:
I've removed the 'return error' but added some commentary below about validations per some of @jsgf 's comments.